sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -423,13 +434,21 @@ case class IntegralDivide(
 
   private lazy val div: (Any, Any) => Any = {
     val integral = left.dataType match {
+      // if it is of Integer type than cast it to Long
+      case _: IntegerType =>
```

Comment:
We need to handle byte/short as well. I think it's easier to update the type coercion rules to do the cast at the analysis time.